File: backend/src/main/java/com/socialmeet/backend/entity/OrderStatus.java
package com.socialmeet.backend.entity;

/**
 * 订单状态枚举
 */
public enum OrderStatus {
    PENDING("pending", "待支付"),
    PROCESSING("processing", "处理中"),
    SUCCESS("success", "成功"),
    PAID("paid", "已支付"),
    EXPIRED("expired", "已过期"),
    CANCELLED("cancelled", "已取消"),
    FAILED("failed", "支付失败");

    private final String code;
    private final String displayName;

    OrderStatus(String code, String displayName) {
        this.code = code;
        this.displayName = displayName;
    }

    public String getCode() {
        return code;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static OrderStatus fromCode(String code) {
        for (OrderStatus status : values()) {
            if (status.code.equals(code)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown order status code: " + code);
    }
}

File: app/src/main/java/com/example/myapplication/model/OrderStatus.kt
package com.example.myapplication.model

/**
 * 订单状态枚举
 */
enum class OrderStatus(val code: String, val displayName: String) {
    PENDING("pending", "待支付"),
    PROCESSING("processing", "处理中"),
    SUCCESS("success", "成功"),
    PAID("paid", "已支付"),
    EXPIRED("expired", "已过期"),
    CANCELLED("cancelled", "已取消"),
    FAILED("failed", "支付失败");
    
    companion object {
        /**
         * 根据代码获取订单状态
         */
        fun fromCode(code: String): OrderStatus? {
            return values().find { it.code == code }
        }
        
        /**
         * 根据代码获取订单状态，如果找不到则返回默认值
         */
        fun fromCodeOrDefault(code: String, default: OrderStatus = PENDING): OrderStatus {
            return fromCode(code) ?: default
        }
    }
}

